src/main/java/seedu/address/newstorage/JsonRecipeBookStorage.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.newstorage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.FileUtil;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.model.ReadOnlyRecipeBook;
+/**
+ * A class to access RecipeBook data stored as a json file on the hard disk.
+ */
+public class JsonRecipeBookStorage implements RecipeBookStorage {
+
+    private static final Logger logger = LogsCenter.getLogger(JsonRecipeBookStorage.class);
+
+    private Path filePath;
+
+    public JsonRecipeBookStorage(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    public Path getRecipeBookFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyRecipeBook> readRecipeBook() throws DataConversionException {
+        return readRecipeBook(filePath);
+    }
+
+    /**
+     * Similar to {@link #readRecipeBook()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataConversionException if the file is not in the correct format.
+     */
+    public Optional<ReadOnlyRecipeBook> readRecipeBook(Path filePath) throws DataConversionException {
+        requireNonNull(filePath);
+
+        Optional<JsonSerializableRecipeBook> jsonRecipeBook = JsonUtil.readJsonFile(
+                filePath, JsonSerializableRecipeBook.class);
+        if (!jsonRecipeBook.isPresent()) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(jsonRecipeBook.get().toModelType());
+        } catch (IllegalValueException ive) {
+            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            throw new DataConversionException(ive);
+        }
+    }
+
+    @Override
+    public void saveRecipeBook(ReadOnlyRecipeBook recipeBook) throws IOException {
+        saveRecipeBook(recipeBook, filePath);
+    }
+
+    /**
+     * Similar to {@link #saveRecipeBook(ReadOnlyRecipeBook)}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     */
+    public void saveRecipeBook(ReadOnlyRecipeBook recipeBook, Path filePath) throws IOException {
```

Comment:
Javadocs missing @param recipeBook.

Reply:
I think it's fine because its similar to the overriden method above?

Reply:
im not sure about this part either because it's referencing to the method above which is overriden